### DISCUSSION TITLE
connect Homeassistant

### DISCUSSION THREAD

**Discussion by scherrmat** [scs/smartmeter-datacollector#57] [Q&A] [answered]
Dear Community

The manual worked fine and I have now a Pi3+ running with the Cable for "Iskaemeco AM550 G3". Grafana Dashboard works fine. Thanks for the detailed documentation for me as a dummy :-)
Now I want to get the readings to my Homeassistant running on a different device in the network. How can this be done? Tried to connect with the MQTT integration, but it seems that in your config, MQTT is only on the local device and port not opened. 

Could someone help me here? Would be great to show the engergy consumption in the energy dashboard of homeassistant. 

Thanks and kind regards
Mathias

**Comment by raymar9** [comment]
Hi Mathias
Thanks a lot for your feedback!
I like the idea of connecting the smartmeter-datacollector with HomeAssistant because this project is very popular at the moment and would be suitable for a sink of the gathered energy data as alternative to the influxdb / Grafana demo.
Currently the only way is though a MQTT broker as you discovered yourself. You can either use a independent MQTT broker that you already use for other purposes or connect the datacollector with the one from our demo.

In our demo we use mosquitto as MQTT broker with the default config. I assume that means that the MQTT port is only open for `localhost` as you found out by yourself. You can change this in the mosquitto configuration (`/etc/mosquitto/mosquitto.conf` or `/etc/mosquitto/conf.d/..`). Check the mosquitto documentation for that.

Currently we do not support the HomeAssistant feature [Auto-Discovery](https://www.home-assistant.io/integrations/mqtt/#mqtt-discovery). However you can manually add MQTT items to HomeAssistant following [this guide](https://www.home-assistant.io/integrations/mqtt/#manual-configured-mqtt-items). Select a suitable MQTT component of the drop-down list. You can get the available MQTT topics with `mosquitto_sub` or by inspecting the influxDB in Grafana.

Please note, I haven't tried this by myself. The above is what I got from documentation.

**Comment by scherrmat** [comment, marked as answer]
**Home Assistant integration - Step by Step**

For all, having the same problem, this How-To might help.
Here is my situation:

- ISKRA AM550 Smartester
- Domoticz adapter from Smartmeter to Raspberry: https://www.amazon.de/dp/B07JW75BZ7/ref=pe_27091401_487027711_TE_SCE_dp_1?th=1
- Rhaspberry Pi3+ installed according to EKZ manual
- Synology DS218+ with Homeassistant Container installed over Docker

**Good to know for the Raspberry:** 
- When going through the manual from EKZ make sure you enable SSH to your Raspi to connect afterwards to the system
- Follow all the steps to set-up the system as described in the manual
- Log in then via any device via SSH (Open terminal, type: ssh username@your raspy -> you are logged in into your Raspberry)
- Enable remote access to Mosquito Broker (I’ve used that manual):
    - https://randomnerdtutorials.com/how-to-install-mosquitto-broker-on-raspberry-pi/
    - type: sudo nano /etc/mosquitto/mosquitto.conf and make the changes as described in manual above
- Now go to your Home-Assistant page. (Einstellungen -> Integration hinzufügen) choose MQTT.
- Choose there your server, use port 1883 and username and password of your mosquitto broker in your raspberry.
- Test connection: „auf ein topic hören“ -> enter # and klick „anfangen zuzuhören“. You should See the messages coming in from the smart meter.
- Next thing is to edit the configuration.yaml file. You will find it in the explorer where home assistant is configured. On my Synology drive it was in the file station docker/homeassistant/ folder

**Modify Configuration.YAML**
Add the following code into the YAML file (can be done with every text editor). Make sure the spaces and „-„ are on the correct place. Replace ISK1xxxxxxxxxxxxx8 with the serial number from your smart meter (as seen in the „auf ein topic hören“ part above)

For tracking the import and export of energy, you would only need „Active_Energy_P_T1“ and „Active_Energy_P_T2“. (Nieder- und Hochtarif). (you could also delete all the other names below if you only want to monitor consumption)

-------paste this into your configuration.yaml file at the end of the file.-------
```python

 # Connection to EKZ Smartmeter
# Example configuration.yaml entry

mqtt:
  sensor:
    - name: "Active_Power_P"
      unique_id: Momentan_aktive_Importleistung_(+A)
      state_topic: "smartmeter/ISK1xxxxxxxxxxxx8/ACTIVE_POWER_P"
      unit_of_measurement: "W"
      device_class: energy
      value_template: "{{ value_json.value }}"
    - name: "Active_Power_N"
      unique_id: Momentan_aktive_Exportleistung_(-A)
      state_topic: "smartmeter/ISK1xxxxxxxxxxxx8/ACTIVE_POWER_N"
      unit_of_measurement: "W"
      device_class: energy
      value_template: "{{ value_json.value }}"
    - name: "Reactive_Power_P"
      unique_id: Momentan_reaktive_Importleistung_(-R)_Blindleistung
      state_topic: "smartmeter/ISK1xxxxxxxxxxxx8/REACTIVE_POWER_P"
      unit_of_measurement: "VAr"
      device_class: energy
      value_template: "{{ value_json.value }}"
    - name: "Reactive_Power_N"
      unique_id: Momentan_reaktive_Importleistung_(+R)_Blindleistung
      state_topic: "smartmeter/ISK1xxxxxxxxxxxx8/REACTIVE_POWER_N"
      unit_of_measurement: "VAr"
      device_class: energy
      value_template: "{{ value_json.value }}"
    - name: "Active_Energy_P"
      unique_id: EKZ_Zaehlerstand_Bezug_total
      state_topic: "smartmeter/ISK1xxxxxxxxxxxx8/ACTIVE_ENERGY_P"
      unit_of_measurement: "Wh"
      device_class: energy
      state_class: total_increasing
      value_template: "{{ value_json.value }}"
    - name: "Active_Energy_P_T1"
      unique_id: EKZ_Zaehlerstand_Bezug_total_Hochtarif
      state_topic: "smartmeter/ISK1xxxxxxxxxxxx8/ACTIVE_ENERGY_P_T1"
      unit_of_measurement: "Wh"
      device_class: energy
      state_class: total_increasing
      value_template: "{{ value_json.value }}"
    - name: "Active_Energy_P_T2"
      unique_id: EEKZ_Zaehlerstand_Bezug_total_Niedertarif
      state_topic: "smartmeter/ISK1xxxxxxxxxxxx8/ACTIVE_ENERGY_P_T2"
      unit_of_measurement: "Wh"
      device_class: energy
      state_class: total_increasing
      value_template: "{{ value_json.value }}"
    - name: "Active_Energy_N"
      unique_id: EKZ_Zaehlerstand_Rueckspeisung_total
      state_topic: "smartmeter/ISK1xxxxxxxxxxxx8/ACTIVE_ENERGY_N"
      unit_of_measurement: "Wh"
      device_class: energy
      state_class: total_increasing
      value_template: "{{ value_json.value }}"
    - name: "Active_Energy_N_T1"
      unique_id: EKZ_Zaehlerstand_Rueckspeisung_total_Hochtarif
      state_topic: "smartmeter/ISK1xxxxxxxxxxxx8/ACTIVE_ENERGY_N_T1"
      unit_of_measurement: "Wh"
      device_class: energy
      state_class: total_increasing
      value_template: "{{ value_json.value }}"
    - name: "Active_Energy_N_T2"
      unique_id: EKZ_Zaehlerstand_Rueckspeisung_Niedertarif
      state_topic: "smartmeter/ISK1xxxxxxxxxxxx8/ACTIVE_ENERGY_N_T2"
      unit_of_measurement: "Wh"
      device_class: energy
      state_class: total_increasing
      value_template: "{{ value_json.value }}"
    - name: "Reactive_Energy_Q1"
      unique_id: Blindenergie_QI_(+Ri)
      state_topic: "smartmeter/ISK1xxxxxxxxxxxx8/REACTIVE_ENERGY_Q1"
      unit_of_measurement: "VArH"
      device_class: energy
      value_template: "{{ value_json.value }}"
    - name: "Reactive_Energy_Q2"
      unique_id: Blindenergie_QII_(+Ri)
      state_topic: "smartmeter/ISK1xxxxxxxxxxxx8/REACTIVE_ENERGY_Q2"
      unit_of_measurement: "VArh"
      device_class: energy
      value_template: "{{ value_json.value }}"
    - name: "Reactive_Energy_Q3"
      unique_id: Blindenergie_QIII_(+Ri)
      state_topic: "smartmeter/ISK1xxxxxxxxxxxx8/REACTIVE_ENERGY_Q3"
      unit_of_measurement: "VArh"
      device_class: energy
      value_template: "{{ value_json.value }}"
    - name: "Reactive_Energy_Q4"
      unique_id: Blindenergie_QIV_(+Ri)
      state_topic: "smartmeter/ISK1xxxxxxxxxxxx8/REACTIVE_ENERGY_Q4"
      unit_of_measurement: "VArh"
      device_class: energy
      value_template: "{{ value_json.value }}"
    - name: "Power_Factor"
      unique_id: Leistungsfaktor
      state_topic: "smartmeter/ISK1xxxxxxxxxxxx8/POWER_FACTOR"
      unit_of_measurement: "cosphi"
      device_class: energy
      value_template: "{{ value_json.value }}"
```
----------
Save the Yaml file and go back to Homeassistant. 
Go to „Entwickerwerkzeuge“ and klick „Konfiguration prüfen“ in the YAML tab. If everything works fine, you should see a green message. 
Restart then Home Assistant. 

Now go to the Energy Dashboard and add the Source „active_energy_p_T1“ and „active_energy_p_T2“. 
You can additionally add also the costs of T1 and T2 (CHF in KWH). 

Klick Save - Done. 
It needs approx 2 hours until the values are displayed.

I hope it works for all of you having the same problem I had.
